scripts/hierarchy.py

```diff
@@ -59,8 +73,10 @@ def node_explorer(graph, node, path):
 
 def graph_explorer(graph):
     for root in graph.roots:
+        root.nutrition = node_nutrition(graph, root)
         yield root
         for node in node_explorer(graph, root, [root.id]):
+            node.nutrition = node_nutrition(graph, node)
```

Comment:
This duplicative call to the `node_nutrition` method, and direct assignment of a member variable on a class object, are both slightly ugly code patterns.  It'd be nice to find a cleaner way to attach the nutritional metadata.

Reply:
(improved; perhaps not ideal, but better than it was)